tests/test_config_parsing/test_parser_error_collection.py

```diff
@@ -0,0 +1,214 @@
+
+                if not content.is_executable:
+                    os.chmod(other_filename, stat.S_IREAD)
+
+    try:
+        ErtConfig.from_file(filename, use_new_parser=True)
+    except ConfigValidationError as err:
+        collected_errors = err.errors
+
+        # Find errors in matching file
+        errors_matching_filename = (
+            [err for err in collected_errors if expected_filename in err.filename]
+            if expected_filename is not None
+            else collected_errors
+        )
+
+        if expected_filename is not None:
+            assert (
+                len(errors_matching_filename) > 0
+            ), f"Expected minimum 1 error matching filename {expected_filename}, got 0"
+        else:
+            assert (
+                len(errors_matching_filename) > 0
+            ), "Expected minimum 1 collected error, got 0"
+
+    def equals_or_expected_any(actual, expected):
+        return True if expected is None else actual == expected
+
+    # Find errors matching location in file
+    errors_matching_location = [
+        x
+        for x in errors_matching_filename
+        if equals_or_expected_any(x.line, expected_line)
+        and equals_or_expected_any(x.column, expected_column)
+        and equals_or_expected_any(x.end_column, expected_end_column)
+    ]
+
+    assert len(errors_matching_location) > 0, (
+        f"Expected to find at least 1 error matching location"
+        f"(line={'*' if expected_line is None else expected_line},"
+        f"column={'*' if expected_column is None else expected_column},"
+        f"end_column{'*' if expected_end_column is None else expected_end_column})"
+    )
+
+    if match is not None:
+        with pytest.raises(ConfigValidationError, match=match):
+            raise ConfigValidationError(errors_matching_location)
+
+
+@dataclass
+class ExpectedErrorInfo:
+    expected_line: Optional[int] = None
+    expected_column: Optional[int] = None
+    expected_end_column: Optional[int] = None
+    expected_filename: str = "test.ert"
+    filename: str = "test.ert"
+    other_files: Dict[str, Union[str, FileDetail]] = None
+    match: Optional[str] = None
+
+
+def assert_multiple_errors_from_config_with(
+    contents: str,
+    expected_errors: List[ExpectedErrorInfo],
+    write_after: Callable[[TextIO], None] = None,
+):
+    for info in expected_errors:
+        assert_error_from_config_with(
+            contents=contents,
+            expected_line=info.expected_line,
+            expected_column=info.expected_column,
+            expected_end_column=info.expected_end_column,
+            expected_filename=info.expected_filename,
+            filename=info.filename,
+            other_files=info.other_files,
+            match=info.match,
+            write_after=write_after,
+        )
+
+
+# "-4 -5" is parsed as the final arg, intended behavior?...
+@pytest.mark.usefixtures("use_tmpdir")
+def test_info_disallowed_kw():
```

Comment:
Would probably be better to use a full sentence ala. `test_that...` as the test name here.